app/src/org/odk/collect/android/jr/extensions/ImageRestrictionExtension.java

```diff
@@ -0,0 +1,44 @@
+package org.odk.collect.android.jr.extensions;
+
+import org.javarosa.core.model.QuestionDataExtension;
+import org.javarosa.core.util.externalizable.DeserializationException;
+import org.javarosa.core.util.externalizable.ExtUtil;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+/**
+ * Represents an extension to the "upload" question type, for setting a maximum
+ * dimension for the size of an image capture
+ *
+ * @author amstone
+ */
+public class ImageRestrictionExtension implements QuestionDataExtension {
```

Comment:
I think we can likely move this straight into /javarosa/ actually, since j2me could also in theory be extended with this info.